### ISSUE TITLE
Extracting from stream fails for Zip format

### ISSUE THREAD

**Issue by princeoffoods** [squid-box/SevenZipSharp#12]
Hello, this can be reproduced in ExtractionFromStreamTest():

//var bytes = File.ReadAllBytes(@"TestData\multiple_files.7z");
var bytes = File.ReadAllBytes(@"TestData\zip.zip");

**Comment by squid-box** [comment]
Running the ExtractionFromStreamTest() pointing to `zip.zip` will give me the following error (is that what you're seeing too?)

```
using (var tmp = new SevenZipExtractor(File.OpenRead(@"TestData\zip.zip")))
{
    tmp.ExtractArchive(OutputDirectory);
}
```

> SevenZip.SevenZipArchiveException : Invalid archive: open/read error! Is it encrypted and a wrong password was provided?
> If your archive is an exotic one, it is possible that SevenZipSharp has no signature for its format and thus decided it is TAR by mistake.

**Comment by princeoffoods** [comment]
Yes, exactly. 
Thanks for looking into this.
The format type is recognized correctly as Zip but then something seems to go wrong, hm.

**Comment by squid-box** [comment]
I'm digging a little into it, but won't be able to fix it now.

The error comes from the `_fileInfo` object in `ArchiveOpenCallback.cs` being null, so when we try to read the filename from the archive it will get a NullReferenceException.

**Comment by squid-box** [comment]
Fix included with release 1.1.53, should appear on Nuget.org shortly.
